spec/compiler/codegen/arithmetics_spec.cr

```diff
@@ -211,6 +211,23 @@ describe "Code gen: arithmetic primitives" do
         end
       {% end %}
 
+      {% if [UInt16, UInt32, UInt64].includes?(type) %}
```

Comment:
Was there a reason to exclude `UInt8` and `UInt128` from this list?

Reply:
`UInt8` is because the issue isn't concerned with same-rank conversions. `UInt128` is because I'm not quite clear how much support for 128-bit integers is actually there (especially for CI).

Reply:
I see. It bugs me slightly that we are not covering all the cases despite the implementation.

I would express that any unsigned int should raise on `-1_{{type}}` for all signed types. For this kind of specs I started `spec/support/number.cr`.

Let us know if you want to tweak these specs further if this makes sense to you. Otherwise it might come in another PR at some point.